### PR TITLE
Upgrade SL30 migration to 1.6

### PR DESCRIPTION
**Pull request by ylavoie**
1. Remove debugging leftover warning
2. Not all settings are defined in SL or previous LSMB databases . Make sure we do not fail migrations in that case.
3. SL didn't have uniqueness requirements for invoices. Suggest adjusted names to ease migration
4. Adjust SL30 migration code for 1.6

### CODE REVIEW

File: lib/LedgerSMB/Upgrade_Tests.pm [hunk 1]
```diff
@@ -703,20 +703,21 @@ push @tests, __PACKAGE__->new(
...
 push @tests, __PACKAGE__->new(
-    test_query => "select *
-                     from ap
-                    where invnumber in (select invnumber
-                                               from ap
-                                              group by invnumber
-                                              having count(*) > 1)
-                    order by invnumber",
-    display_name => $locale->text('Non-unique invoice numbers'),
+    test_query => "SELECT id, concat(invnumber,'-',row_number() over(partition by invnumber order by id)) AS invnumber,
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Upgrade_Tests.pm]: @ylavoie with this change, we no longer show the user the problematic content. I'd expect that to be confusing more than that this commit is helpful (in the sense that this commit prevents data-entry)

File: lib/LedgerSMB/Upgrade_Tests.pm [hunk 2]
```diff
@@ -703,20 +703,21 @@ push @tests, __PACKAGE__->new(
...
+                                        HAVING count(*) > 1)
+                 ORDER BY invnumber",
+    display_name => $locale->text('Non-unique invoice numbers detected'),
     name => 'no_duplicate_ap_invoicenumbers',
     display_cols => ['id', 'invnumber', 'transdate', 'duedate', 'datepaid',
                      'ordnumber', 'quonumber', 'approved'],
     column => 'invnumber',
  instructions => $locale->text(
-                   'Please make all AP invoice numbers unique'),
+                   'Please review suggestions to make all AP invoice numbers unique'),
```
Comment (ehuelsmann) [inline lib/LedgerSMB/Upgrade_Tests.pm]: Maybe we should list the conflicting data as well as the suggested data and explain something about that in the text above?

File: lib/LedgerSMB/Upgrade_Tests.pm (hunk 1 above)
Comment (ylavoie) [inline lib/LedgerSMB/Upgrade_Tests.pm]: We still show all problematic entries, but added a suffix to make them unique.

Comment (ehuelsmann) [inline lib/LedgerSMB/Upgrade_Tests.pm]: Sure, but the entries don't look problematic anymore...

**Comment by ylavoie** [comment]
This depends on #2288

**Pull request merged**